SAN/code/network_simulations/run_batch_benchmark.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# first argument $1 is the name of the julia file 
+# second argument $2 is the address of the s3 container to transfer file
+# third argument $3 is the number of nodes
+
+# setting up access (only needed on first use)
+aws s3 cp s3://ran-s3-install-pkgs/config/RanPocKP.pem .
+chmod 400 RanPocKP.pem
+
+# copy julia script to s3
+aws s3 cp "$1" "$2"
+
+sudo su - ec2-user <<ENDSUDO
+ssh -o StrictHostKeyChecking=no -i RanPocKP.pem ec2-user@172.31.100.6 <<ENDSSH
+aws s3 cp "$2""$1" ~/"$1"
+#nohup time julia ~/"$1">output_"$1".txt
+#nohup time env JULIA_NUM_THREADS=4 julia ~/"$1">output_"$1".txt #4 threads
+#nohup time julia -p 4 ~/"$1">output_"$1".txt  # 4 workers
+nohup time julia ~/"$1">output_"$3"_"$1".txt $3 # calling julia
+aws s3 cp /home/ec2-user/output_"$3"_"$1".txt "$2"output_"$3"_"$1".txt #copy file over
+aws s3 cp /home/ec2-user/data_out_"$3".csv "$2"data_out_"$3".csv
+aws s3 cp /home/ec2-user/variables_"$3".jld2 "$2"variables_"$3".jld2
+aws s3 cp /home/ec2-user/profile_"$3".txt "$2"profile_"$3".txt
```

Comment:
Yes If profile_$N.txt does not exist then it just errors and proceeds on, so it's not a big issue. However, if profile_$N.txt exists from a previous run it would copy it. Unsure how to solve it, but good to note the issue.

Reply:
One easy way is to create an empty profile_$N.txt file at the beginning?